### PR TITLE
Issue 307 - Added Save

### PR DESCRIPTION
**Pull request by f-prime**
This turned into a big PR because there was a lot that needed to be changed that I didn't anticipate going into this. Here is the outline of what this contains:

1. A saves table has been added to the database that is nearly identical to the likes table

2. Added save/unsave resolvers as well as GraphQL mutations for both of them (works nearly identically to the current like/unlike resolvers and mutations)

3. Added getIsIdeaSavedByMe query 

4. Generalized the `get_liked_ideas` lambda function to `get_acted_on_ideas` which will return either liked or saved ideas depending on the env variables passed to it which are set based on whether the `getSavedIdeas` or `getLikedIdeas` query is called

5. Generalized the `LikeIdea.vue` component to `ActOnIdea.vue` which supports both saving and liking actions. I did try separating out the two actions into their own components but found that the functionalities were so similar that it made more sense to me to keep them together.

### CODE REVIEW

**Comment by ezl** [comment]
@f-prime i can see this already has conflicts -- can you first resolve the conflicts? This is a big PR, so it'll be good to clear away any obvious barriers to merging up front.

File: front/components/ideaDetail/ActOnIdea.vue [hunk 1]
```diff
@@ -80,49 +111,58 @@ export default {
...
+    },
+
+    emitStateChange(result) {
+      if(this.action === 'like') {
+        this.$emit('likedStateChanged', {
+          liked: this.isActedOn,
+          likesCount: result.likesCount
+        })
+      } else {
+        this.$emit('savedStateChanged', {
```
Comment (aldarund) [inline front/components/ideaDetail/ActOnIdea.vue:125]: emit should be dash cased see vue guide

**Comment by aldarund** [comment]
I not quite like this approach that we do if else inside component for idea/like.  Better to extract common part into new component and pass there options like mutation, icon etc and than create two component - like and save and use them instead of passing action=like/save,

**Comment by f-prime** [comment]
- Converted event names to kebab case
- Fixed merge conflicts
- Created LikeIdea and SaveIdea components

**Pull request merged**